lgc/state/PalMetadata.cpp

```diff
@@ -245,11 +245,48 @@ void PalMetadata::setUserDataSpillUsage(unsigned dwordOffset) {
 // TODO Shader compilation: The idea is that this will be called at the end of a pipeline compilation, or in
 // an ELF link, but not at the end of a shader/half-pipeline compile.
 void PalMetadata::finalizePipeline() {
+  // Set pipeline hash.
+  auto pipelineHashNode = m_pipelineNode[Util::Abi::PipelineMetadataKey::InternalPipelineHash].getArray(true);
+  const auto &options = m_pipelineState->getOptions();
+  pipelineHashNode[0] = m_document->getNode(options.hash[0]);
+  pipelineHashNode[1] = m_document->getNode(options.hash[1]);
+
+  // Set PA_CL_CLIP_CNTL from pipeline state settings.
+  bool depthClipDisable = !m_pipelineState->getViewportState().depthClipEnable;
+  uint8_t usrClipPlaneMask = m_pipelineState->getRasterizerState().usrClipPlaneMask;
+  bool rasterizerDiscardEnable = m_pipelineState->getRasterizerState().rasterizerDiscardEnable;
+  PA_CL_CLIP_CNTL value = {};
```

Comment:
value -> paClClipCntl. I prefer to use this. Once there are multiple registers, you have to rename `value`